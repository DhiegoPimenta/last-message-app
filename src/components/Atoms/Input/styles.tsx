import styled, { css } from 'styled-components'

export const Input = styled.div`
    ${({ theme }) => css`
        border-radius: ${theme.border.radius};
        border: 1px solid ${theme.colors.gray};
        display: flex;
        width: 45rem;
        height: 5rem;
        padding: 0 ${theme.spacings.small};
        align-items: center;

        input {
            height: 100%;
            width: 100%;
            border: none;
            margin: 0 ${theme.spacings.xsmall};
            outline: none;
            font-size: ${theme.font.sizes.small};

            ::placeholder {
                color: ${theme.colors.gray};
                font-size: ${theme.font.sizes.small};
            }
        }
    `}
`

export const ErrorMessage = styled.span`
    ${({ theme }) => css`
        color: ${theme.colors.danger};
        font-size: ${theme.font.sizes.xsmall};
        margin-top: ${theme.spacings.xxsmall};
    `}
`
