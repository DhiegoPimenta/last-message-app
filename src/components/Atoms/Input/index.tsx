import { HTMLInputTypeAttribute } from 'react'
import Icon, { IconsNames } from '../Icon'
import * as S from './styles'

export interface InputProps
    extends React.ButtonHTMLAttributes<HTMLInputElement> {
    iconName?: IconsNames
    errorMessage?: string
    inputType?: HTMLInputTypeAttribute
    handleChange?(text: string): void
}

const Input = ({
    iconName,
    errorMessage,
    inputType = 'text',
    handleChange,
    ...props
}: InputProps) => {
    return (
        <>
            <S.Input>
                {iconName && <Icon iconName={iconName} size="small" />}
                <input
                    type={inputType}
                    {...props}
                    onChange={(e: React.FormEvent<HTMLInputElement>) =>
                        handleChange && handleChange(e.currentTarget.value)
                    }
                />
            </S.Input>
            <S.ErrorMessage>{errorMessage}</S.ErrorMessage>
        </>
    )
}

export default Input
