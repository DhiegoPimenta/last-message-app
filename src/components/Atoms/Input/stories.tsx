import { Story, Meta } from '@storybook/react/types-6-0'
import Input, { InputProps } from '.'

export default {
    title: 'Atoms/Input',
    component: Input,
} as Meta

export const Basic: Story<InputProps> = (args) => <Input {...args} />
Basic.args = {
    iconName: 'mail',
    errorMessage: 'error message teste',
}
