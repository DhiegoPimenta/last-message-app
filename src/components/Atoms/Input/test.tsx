import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import Input from '.'

describe('<Input/>', () => {
    it('Should be render Input', () => {
        const { container } = renderWithTheme(<Input iconName="home" />)

        expect(screen.getByLabelText('Icone de casa')).toBeInTheDocument()
        expect(container.firstChild).toMatchSnapshot()
    })
})
