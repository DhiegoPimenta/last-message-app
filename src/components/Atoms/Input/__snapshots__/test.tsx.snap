// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> Should be render Input 1`] = `
.c1 {
  width: 3.5rem;
  height: 3.5rem;
}

.c0 {
  border-radius: 1rem;
  border: 1px solid #CECECE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 45rem;
  height: 5rem;
  padding: 0 2.4rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 input {
  height: 100%;
  width: 100%;
  border: none;
  margin: 0 1.6rem;
  outline: none;
  font-size: 1.8rem;
}

.c0 input::-webkit-input-placeholder {
  color: #CECECE;
  font-size: 1.8rem;
}

.c0 input::-moz-placeholder {
  color: #CECECE;
  font-size: 1.8rem;
}

.c0 input:-ms-input-placeholder {
  color: #CECECE;
  font-size: 1.8rem;
}

.c0 input::placeholder {
  color: #CECECE;
  font-size: 1.8rem;
}

<div
  class="c0"
>
  <img
    aria-label="Icone de casa"
    class="c1"
    src="/img/icons/home.png"
  />
  <input />
</div>
`;
